runelite-client/src/main/java/net/runelite/client/plugins/bank/BankItemsOverlay.java

```diff
@@ -0,0 +1,166 @@
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client.plugins.bank;
+
+import lombok.NonNull;
+import lombok.Value;
+import lombok.extern.slf4j.Slf4j;
+import net.runelite.api.Item;
+import net.runelite.client.game.ItemManager;
+import net.runelite.client.ui.overlay.Overlay;
+import net.runelite.client.ui.overlay.OverlayPosition;
+import net.runelite.client.ui.overlay.components.ComponentOrientation;
+import net.runelite.client.ui.overlay.components.ImageComponent;
+import net.runelite.client.ui.overlay.components.PanelComponent;
+import net.runelite.client.ui.overlay.components.TitleComponent;
+import net.runelite.client.util.StackFormatter;
+
+import javax.inject.Inject;
+import java.awt.Color;
+import java.awt.Dimension;
+import java.awt.Graphics2D;
+import java.awt.Point;
+import java.awt.image.BufferedImage;
+import java.util.Arrays;
+import java.util.Comparator;
+
+@Slf4j
+class BankItemsOverlay extends Overlay
+{
+	private static final int ITEM_SIZE = 10;
+	private static final int PLACEHOLDER_WIDTH = 36;
+	private static final int PLACEHOLDER_HEIGHT = 32;
+	private static final int WRAPPING = 2;
+	private static final Point GAP = new Point(10, 6);
+	private static final ImageComponent PLACEHOLDER_IMAGE = new ImageComponent(new BufferedImage(PLACEHOLDER_WIDTH, PLACEHOLDER_HEIGHT, BufferedImage.TYPE_4BYTE_ABGR));
+
+	private final PanelComponent panelComponent = new PanelComponent();
+
+	private final ItemManager itemManager;
+	private final BankConfig config;
+	private final BankCalculation bankCalculation;
+
+
+	private ItemPrice[] expensiveItems;
+
+	@Inject
+	private BankItemsOverlay(ItemManager itemManager, BankConfig config, BankCalculation bankCalculation)
+	{
+		setPosition(OverlayPosition.BOTTOM_RIGHT);
+		panelComponent.setWrapping(WRAPPING);
+		panelComponent.setGap(GAP);
+		panelComponent.setOrientation(ComponentOrientation.HORIZONTAL);
+		panelComponent.setPreferredSize(new Dimension(PLACEHOLDER_WIDTH + GAP.x, PLACEHOLDER_HEIGHT + GAP.y));
+
+		this.itemManager = itemManager;
+		this.config = config;
+		this.bankCalculation = bankCalculation;
+	}
+
+	@Override
+	public Dimension render(Graphics2D graphics)
+	{
+		if (!config.showMostExpensiveItemsOverlay())
+		{
+			return null;
+		}
+
+		final Item[] bankItems = bankCalculation.getBankItems();
+
+		if (bankItems == null || bankItems.length == 0)
+		{
+			return null;
+		}
+
+		panelComponent.getChildren().clear();
+
+		if (expensiveItems == null || bankCalculation.isBankDifferent(bankItems)) 
+		{
+			expensiveItems = getExpensiveItems(bankItems);
+			Arrays.sort(expensiveItems, Comparator.comparing(ItemPrice::getPrice).reversed());
+		}
+
+
+		for (int i = 0; i < ITEM_SIZE; i++)
```

Comment:
Why does maintaining a 10 item size matter?

Reply:
It doesn't. It's just a random number I chose, and mentioned it can be configurable if we thought appropriate.

Some of this logic is from how I saw it was happening in `InventoryViewerOverlay`, so I carried it over. I decided to keep the idea of having a non-resizing panel, which is what this (and the placeholder additions below) tries to accomplish.

If you can think of a more intuitive design, I'd be more than happy to change it.